### PR TITLE
until

### LINKED ISSUE #1690: until
Like `R.unless` but applies function until predicate is satisfied and passes new values to predicate each step.

Or `while` which is the same but with `complement` pred

```js
R.until(
    R.gt(R.__, 100),
    R.multiply(2)
)(1) === 128
```

### PR DESCRIPTION
**Pull request by iofjuupasli**
fixes #1690

English is not my native, and so I'm not sure what is clear description for this function.

### CODE REVIEW

File: src/until.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * @category Logic
+ * @sig (a -> Boolean) -> (a -> a) -> a -> a
+ * @param {Function} pred A predicate function
+ * @param {Function} fn The iterator function
+ * @param {*} init Initial value
+ * @return {*} Final value that satisfies predicate
+ * @example
+ *
+ *      R.until(R.gt(R.__, 100), R.multiply(2))(1) // => 128
+ *      R.until(R.none(R.isArrayLike), R.unnest)([1, [2], [[3]]]) // => [1, 2, 3]
```
Comment (davidchambers) [inline src/until.js]: Let's remove this example as [`R.flatten`][1] would be much clearer. ;)

The first example is great!


[1]: http://ramdajs.com/docs/#flatten

**Comment by davidchambers** [comment]
For the description, I suggest something like this:

_Takes a predicate, a transformation function, and an initial value, and returns a value of the same type as the initial value. It does so by applying the transformation until the predicate is satisfied, at which point it returns the satisfactory value._

**Comment by CrossEye** [comment]
This is great.  Thank you.

I like David's description, except that I would substitute "Accepts" for "Takes".  I always wonder where functions are taking things and if thievery is involved.  :smile:

I don't know if a additional sentence is in order warning about the possibility of infinite loops.  I know it's possible in other Ramda functions, but this one seems more likely.

**Comment by iofjuupasli** [comment]
Does it need anything else?

**Comment by CrossEye** [comment]
:herb:  I'm happy with this, unless someone wants to do some bikeshedding on the name.

**Comment by buzzdecafe** [comment]
gotta squash the commits then, please

**Comment by CrossEye** [comment]
Thanks, @iofjuupasli!

**Pull request merged**